test-data/unit/fine-grained-attr.test

```diff
@@ -21,3 +21,28 @@ class A:
 [out]
 ==
 main:5: error: Incompatible return value type (got "Attribute[float]", expected "Attribute[int]")
+
+[case magicAttributeConsistency]
+import m
+
+[file c.py]
+from attr import define
+
+@define
+class A:
+    a: float
+    b: int
+[builtins fixtures/attr.pyi]
+
+[file m.py]
+from c import A
+
+A.__attrs_attrs__.a
+
+[file m.py.2]
+from c import A
+
+A.__attrs_attrs__.b
```

Comment:
I think in this case you can also add `b` field for `.2` only. It will make more sense.

Reply:
Hm, can you clarify?

Reply:
Something like

```ini
[case magicAttributeConsistency]
import m

[file c.py]
from attr import define

@define
class A:
    a: float

[file c.py.2]
from attr import define

@define
class A:
    a: float
    b: int
[builtins fixtures/attr.pyi]

[file m.py]
from c import A

A.__attrs_attrs__.a

[file m.py.2]
from c import A

A.__attrs_attrs__.b
```

Reply:
I can add this, but this isn't testing what I'm fixing. I.e. your test, as written, will pass on `master`.

The issue is loading the magic attribute from Mypy cache. Your suggestion has two versions of `c.py`, so the cache will not be used.

Let me know what you think.